app/controllers/application_controller.rb

```diff
@@ -1,31 +1,24 @@
 class ApplicationController < ActionController::Base
   include Pundit
-
   rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
 
-  before_action :ensure_logged_in
-  helper_method :current_user
+  before_action :authenticate_user!
+  before_action :ensure_verified_user!
+
 
-  def ensure_logged_in
-    redirect_to root_path unless current_user
+  def ensure_verified_user!
+    unless current_user && current_user.verified
+      flash[:error] = "Your account is not verified!"
+      sign_out current_user
+    end
   end
 
   def ensure_superuser
-    redirect_to root_path unless current_user && current_user.superuser?
+    redirect_to root_path unless current_user.superuser?
   end
```

Comment:
Shouldn't this be a policy instead of being a global function?

Reply:
Yeah. It would be better to do as policy 🤔
But this was not the motto of this PR 🤔
Creating a new issue would do the need? 🤔